[FIX][14.0] Access rights issue helpdesk_mgmt

aoso3 opened the pull request:
Fix the problem of users with "user" access rights can't access Helpdesk dashboard.

PR:  https://github.com/OCA/helpdesk/pull/222/files

issue #220

dalonsod commented:
@aoso3 could you squash commits and reword to e.g. `[FIX] Dashboard access rights issue for User role` or similar? Thanks for the contribution!

gdgellatly commented:
This is not the correct approach IMO, the real issue is/was writing todo_ticket_ids in the function. In any case compute_sudo is going to introduce a bug, none of the other similar models with similar access rights use it, e.g. check Projects, Journals, CRM Teams. None require compute_sudo as non stored functional fields.

dalonsod commented:
@aoso3 could you take a look at solution provided by @gdgellatly and also applied in #222 ?

Commit:
add_missing_rigths

aoso3 commented:
@dalonsod @gdgellatly 

Sorry for the late reply.

I have removed the use of "compute_sudo" and added the missing rights to the view, can you please check the PR again.

Thank you.

gdgellatly commented:
@aoso3 It might pay to check, I think we fixed this ages ago

vongoethe reviewed (approved):
please merge

pedrobaeza commented:
It was fixed in #171, but for 13.0, so let's merge it for v14 as well:

/ocabot merge minor

pedrobaeza commented:
As this has a problem with pre-commit, I have cherry-picked the v13 commit at 4f5be853b68496332e7586a2d9e61d69308fd750 and also on v15 at d9f75881d8b108d9c4839d204618bb8d3800d946
